File: UI/barbershop_admin/lib/models/news.dart
import 'package:json_annotation/json_annotation.dart';

part 'news.g.dart';

@JsonSerializable()
class News {
  int? id;
  String? title;
  String? content;
  String? photo;
  DateTime? createdDateTime;
  int? authorId;
  String? authorFullName;

  News({
    this.id,
    this.title,
    this.content,
    this.photo,
    this.createdDateTime,
    this.authorId,
    this.authorFullName,
  });

  factory News.fromJson(Map<String, dynamic> json) => _$NewsFromJson(json);

  Map<String, dynamic> toJson() => _$NewsToJson(this);
}

File: Core/Entities/News.cs
namespace Core.Entities
{
    public class News
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public string Photo { get; set; }
        public DateTime CreatedDateTime { get; set; } = DateTime.UtcNow;
        
        public int? AuthorId { get; set; }
        public virtual AppUser Author { get; set; }
    }
}